### PR TITLE
Feature/#38 load study group from tapi

### PR DESCRIPTION
**Pull request by davidm1766**

### CODE REVIEW

File: Backend/Converter/ConverterApiToDomain.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        {
+            
+            Timetable timetable = new Timetable();
+            
+            for (int idxDay = 0;idxDay < schedule.DaysInWeek.Count;idxDay++)
+            {
+                var maxBlocks = schedule.DaysInWeek[idxDay].BlocksInDay.Count;
+
+                byte startingBlock = 0;
+                for (int blckIdx = 1;blckIdx < maxBlocks;blckIdx++)
```
Comment (ikim23) [inline Backend/Converter/ConverterApiToDomain.cs]: Space after semicolon?

File: Tests/BackendTest/ConverterTest.cs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            day.BlocksInDay.Add(slot11);
+            day.BlocksInDay.Add(slot12);
+            day.BlocksInDay.Add(slot13);
+
+            ScheduleWeekContent week = new ScheduleWeekContent();
+            week.DaysInWeek.Add(day);
+            var timetable = await ConverterApiToDomain.ConvertTimetableForGroup(week, serviceCourse,new FakeProxy());
+
+            timetable.Blocks.Count.Should().Be(6);
+            var blok = timetable.Blocks.Where(x => x.StartHour == 7).FirstOrDefault();
```
Comment (marektavac) [inline Tests/BackendTest/ConverterTest.cs]: Could be simple:
```C#
var blok = timetable.Blocks.FirstOrDefault(x => x.StartHour == 7);
```

File: Tests/BackendTest/ConverterTest.cs [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        [Fact]
+        public async void ConvertTest_ValidStudyGroup()
+        {
+            IMongoDatabase database = _mongoFixture.MongoClient.GetDatabase("StudentsDB");
+            var service = new StudyGroupService(database);
+            var serviceCourse = new CourseService(database);
+
+
+            StudyGroup grp = null;
+            try
```
Comment (marektavac) [inline Tests/BackendTest/ConverterTest.cs]: Why do you have try block here?

File: Tests/BackendTest/ConverterTest.cs [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace BackendTest
+{
+    [Collection("Database collection")]
+    public class ConverterTest : IClassFixture<Mongo2GoFixture>
```
Comment (marektavac) [inline Tests/BackendTest/ConverterTest.cs]: Some not valid tests? what is happend?

File: Backend/Converter/ConverterApiToDomain.cs (hunk 1 above)
Comment (davidm1766) [reply Backend/Converter/ConverterApiToDomain.cs]: ok

File: Backend/Converter/ConverterApiToDomain.cs [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+using FRITeam.Swapify.Backend.Interfaces;
+using FRITeam.Swapify.Entities;
+using FRITeam.Swapify.Entities.Enums;
+
+namespace FRITeam.Swapify.Backend.Converter
+{
+    public class ConverterApiToDomain
+    {
+
+        public async static Task<Timetable> ConvertTimetableForGroup(ScheduleWeekContent schedule, ICourseService courseServ, ISchoolScheduleProxy proxy)
```
[the remark this answers is not in the document]
Comment (davidm1766) [reply Backend/Converter/ConverterApiToDomain.cs]: ok

File: Backend/Converter/ConverterApiToDomain.cs [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+                var downloadedTimetable = proxy.GetBySubjectCode(courseName);
+                var convertedTimetable = await ConvertTimetableForCourse(downloadedTimetable,courseServ,proxy);
+                course = new Course() { CourseName = courseName, Timetable = convertedTimetable };
+                await courseServ.AddAsync(course);
+            }
+            return course.Id;
+            
+        }
+        
+        private static bool IsSameBlock(ScheduleHourContent b1, ScheduleHourContent b2)
```
[the remark this answers is not in the document]
Comment (davidm1766) [reply Backend/Converter/ConverterApiToDomain.cs]: ok

File: Backend/CourseService.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        public CourseService(IMongoDatabase database)
+        {
+            _database = database;
+        }
+
+        public async Task AddAsync(Course entityToAdd)
+        {
+            entityToAdd.Id = Guid.NewGuid();
+            await _database.GetCollection<Course>(nameof(Course)).InsertOneAsync(entityToAdd);
+        }
```
[the remark this answers is not in the document]
Comment (davidm1766) [reply Backend/CourseService.cs]: ok

File: Backend/StudyGroupService.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            await _database.GetCollection<StudyGroup>(nameof(StudyGroup)).InsertOneAsync(entityToAdd);
+        }
+
+        public async Task<StudyGroup> FindByIdAsync(Guid guid)
+        {
+            var collection = _database.GetCollection<StudyGroup>(nameof(StudyGroup));
+            return await collection.Find(x => x.Id.Equals(guid)).FirstOrDefaultAsync();
+        }
+
+        public async Task<StudyGroup> GetStudyGroupAsync(string studyGroupNumber, ICourseService courseServ, ISchoolScheduleProxy proxy)
```
[the remark this answers is not in the document]
Comment (davidm1766) [reply Backend/StudyGroupService.cs]: Ok I moved this method to CourseService

File: Tests/BackendTest/ConverterTest.cs (hunk 2 above)
Comment (davidm1766) [reply Tests/BackendTest/ConverterTest.cs]: ok

File: Tests/BackendTest/ConverterTest.cs (hunk 3 above)
Comment (davidm1766) [reply Tests/BackendTest/ConverterTest.cs]: OK it is no needed

File: Backend/Converter/ConverterApiToDomain.cs [hunk 9]
```diff
@@ -13,36 +13,36 @@ namespace FRITeam.Swapify.Backend.Converter
...
-        public async static Task<Timetable> ConvertTimetableForCourse(ScheduleWeekContent schedule, ICourseService courseServ, ISchoolScheduleProxy proxy)
+        public static async Task<Timetable> ConvertTimetableForCourseAsync(ScheduleWeekContent courseTimetable, ICourseService courseServ, ISchoolScheduleProxy proxy)
         {
-            return await ConvertTimetable(schedule, courseServ, proxy, true);
+            return await ConvertTimetable(courseTimetable, courseServ, proxy, true);
         }
 
 
-        private async static Task<Timetable> ConvertTimetable(ScheduleWeekContent schedule, ICourseService courseServ, ISchoolScheduleProxy proxy,bool isTimetableForCourse)
+        private static async Task<Timetable> ConvertTimetable(ScheduleWeekContent schedule, ICourseService courseServ, ISchoolScheduleProxy proxy, bool isTimetableForCourse)
```
Comment (marektavac) [inline Backend/Converter/ConverterApiToDomain.cs]: Also this one is async....

File: Backend/CourseService.cs [hunk 10]
```diff
@@ -11,6 +13,7 @@ namespace FRITeam.Swapify.Backend
     public class CourseService : ICourseService
     {
         private readonly IMongoDatabase _database;
+        private IMongoCollection<Course> _courseCollection => _database.GetCollection<Course>(nameof(Course));
```
Comment (marektavac) [inline Backend/CourseService.cs]: Also properties we write with _loweCamelCase?

File: Backend/Converter/ConverterApiToDomain.cs (hunk 9 above)
Comment (davidm1766) [reply Backend/Converter/ConverterApiToDomain.cs]: OK

File: Backend/CourseService.cs (hunk 10 above)
Comment (davidm1766) [reply Backend/CourseService.cs]: I use _ for private field and private propetries, but we can discute about convention on wednesday

Comment (marektavac) [reply Backend/CourseService.cs]: So you can merge it for now... Refactor can be later

**Pull request merged**